src/main/java/de/zalando/aruha/nakadi/repository/kafka/KafkaTopicRepository.java

```diff
@@ -197,9 +197,17 @@ public void syncPostBatch(final String topicId, final List<BatchItem> batch) thr
             final boolean isSuccessful = done.await(settings.getKafkaSendTimeoutMs(), TimeUnit.MILLISECONDS);
 
             if (!isSuccessful) {
+                for (final BatchItem item : batch) {
+                    item.safeFail("timed out");
```

Comment:
Why do you set it for each item?
The posible scenario can be that all events except one were submitted and only one timed out and in that case `isSuccessful` will be false.